src/StringUtils.cpp

```diff
@@ -108,6 +108,76 @@ std::pair<std::string, std::string> NAS2D::splitOnLast(const std::string& str, c
 	}
 }
 
+std::vector<std::string> NAS2D::splitOnUnquoted(const std::string& str, char delim /*= ','*/)
+{
+	bool in_quote = false;
+	std::vector<std::string> result{};
+	auto potential_count = 1u + std::count(std::begin(str), std::end(str), delim);
+	result.reserve(potential_count);
+	auto substring_start = std::begin(str);
+	auto substring_end = std::end(str);
+	for (auto iter = std::begin(str); iter != std::end(str); /* DO NOTHING */)
+	{
+		if (*iter == '"')
+		{
+			in_quote = !in_quote;
+			++iter;
+			continue;
+		}
+		if (in_quote || *iter != delim)
```

Comment:
I'm a little uncertain about this suggestion, but....

Here there is a stateful `in_quote` variable, and the nested `if` statements in a loop. Often the state variable can be removed by splitting the loop into two. I suspect that can be done here. One loop to skip over ignored stuff or an opening quote, and another loop to find the end of the current string (close quote, delimiter, end of string). Or perhaps even just two calls to [`std::find`](https://en.cppreference.com/w/cpp/algorithm/find) instead of two loops. Admittedly it's complicated by both delimiters and quotes, and you'd still need an overall loop here to handle multiple strings for the split result.

Reply:
I'm not quite understanding what you are suggesting. Do you have an example?

Reply:
I'm afraid I don't have an example, and would need to defer on finding one. I haven't played around with the details enough to know if the particular concerns here (multiple delimiters: quotes, specified delimiter, end of string) would get in the way of using higher-level methods. If not, it might be possible to use shorter higher-level code. Probably not a high priority item though, so feel free to ignore.